kinto_client/__init__.py

```diff
@@ -169,18 +171,36 @@ def _paginated(self, endpoint, records=None, visited=None):
 
         return records.values()
 
+    def _get_cache_headers(self, safe, data=None, last_modified=None):
+        if (last_modified is None
+                and data is not None and data.get('last_modified')):
+            last_modified = data['last_modified']
+
+        if safe and last_modified is not None:
+            return {'If-Match': utils.quote(last_modified)}
+        # else return None
+
     # Buckets
 
-    def update_bucket(self, bucket=None, permissions=None):
+    def create_bucket(self, bucket=None, data=None, permissions=None,
+                      safe=True):
+        headers = DO_NOT_OVERWRITE if safe else None
         endpoint = self._get_endpoint('bucket', bucket)
         resp, _ = self.session.request('put', endpoint,
-                                       permissions=permissions)
+                                       permissions=permissions,
+                                       headers=headers)
         return resp
 
-    def delete_bucket(self, bucket=None):
+    def update_bucket(self, bucket=None, data=None, permissions=None,
+                      safe=True):
         endpoint = self._get_endpoint('bucket', bucket)
-        resp, _ = self.session.request('delete', endpoint)
-        return resp['data']
+        resp, _ = self.session.request(
+            'patch', endpoint,
+            data=data,
+            permissions=permissions,
+            headers=self._get_cache_headers(safe, data))
```

Comment:
nit: you don't same indentation as for `create_bucket`. Intermediary variable for headers ?